cmd/release-controller/sync.go

```diff
@@ -514,6 +517,124 @@ func (c *Controller) syncReady(release *Release) error {
 	return nil
 }
 
+// GitHubPR contains the necessary info to identify a GitHub PR (org, repo, and pr number)
+type GitHubPR struct {
+	Org  string
+	Repo string
+	PR   int
+}
+
+// BugzillaPR contains a bugzilla ID and the associated GitHub PR that resolves the bug
+type BugzillaPR struct {
+	bugzillaID int
+	githubPR   GitHubPR
+}
+
+var (
+	// GHRegex matches github PR links
+	GHRegex = regexp.MustCompile(`\(https://github.com/[[:alnum:]]+/[[:alnum:]|\-]+/pull/[\d]+\)`)
+	// BZRegex matches the markdown labels to bugzilla links
+	BZRegex = regexp.MustCompile(`\[Bug [\d]+\]`)
+	// BZAssignRegex matches the QA assignment comment made by the openshift-ci-robot
+	BZAssignRegex = regexp.MustCompile(`Requesting review from QA contact:[[:space:]]+/cc @[[:alnum:]]+`)
+	// from prow lgtm plugin
+	lgtmRe       = regexp.MustCompile(`(?mi)^/lgtm(?: no-issue)?\s*$`)
+	lgtmCancelRe = regexp.MustCompile(`(?mi)^/lgtm cancel\s*$`)
+)
+
+// getBugzillaPRs identifies bugzilla bugs and the associated github PRs fixed in a release from
+// a given markdown changelog generated by `oc adm release info --changelog=git-cache-path from-tag to-tag`
+func getBugzillaPRs(input string) []BugzillaPR {
```

Comment:
We should not be doing this - the release info command already has a mode that spits out JSON/structured list of bugs for a release.  So you'll need to add a new info path "get bugs for release changes" and then instead of changelog we'll run `--bugs` with `-o json`.

Reply:
I was using `changelog` because I wanted to retrieve the associated PR as well, which `--bugs` doesn't directly provide. I also assumed that `--bugs` didn't have `json` since `changelog` didn't. That's useful.

Since we add the github PRs as external bugs in bugzilla though, I could identify the github PR via the bugzilla API though. I'll implement it like that.